### PR TITLE
[IOTDB-3448] Migrate the logic of deleteRegion onto the consensus module

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira3448](https://issues.apache.org/jira/browse/IOTDB-3448)

### CODE REVIEW

**Review by qiaojialin — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java [hunk 1]
```diff
@@ -365,22 +363,33 @@ public TSStatus invalidatePermissionCache(TInvalidatePermissionCacheReq req) thr
...
     } else {
-      return SchemaRegionConsensusImpl.getInstance()
-          .write(consensusGroupId, deleteRegionNode)
-          .getStatus();
+      ConsensusGenericResponse response =
+          SchemaRegionConsensusImpl.getInstance().removeConsensusGroup(consensusGroupId);
+      if (!response.isSuccess()
+          && !(response.getException() instanceof PeerNotInConsensusGroupException)) {
+        return RpcUtils.getStatus(
+            TSStatusCode.INTERNAL_SERVER_ERROR, response.getException().getMessage());
```
Comment (MrQuansy) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java:383]: Use DELETE_REGION_ERROR instead.

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java [hunk 2]
```diff
@@ -365,22 +363,33 @@ public TSStatus invalidatePermissionCache(TInvalidatePermissionCacheReq req) thr
...
     if (consensusGroupId instanceof DataRegionId) {
-      return DataRegionConsensusImpl.getInstance()
-          .write(consensusGroupId, deleteRegionNode)
-          .getStatus();
+      ConsensusGenericResponse response =
+          DataRegionConsensusImpl.getInstance().removeConsensusGroup(consensusGroupId);
+      if (!response.isSuccess()
+          && !(response.getException() instanceof PeerNotInConsensusGroupException)) {
+        return RpcUtils.getStatus(
+            TSStatusCode.INTERNAL_SERVER_ERROR, response.getException().getMessage());
```
Comment (MrQuansy) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java:374]: Not use INTERNAL_ERROR.  Maybe choose other detailed TSStatusCode, like DELETE_REGION_ERROR.

**Comment by MrQuansy** [comment]
PTAL @OneSizeFitsQuorum

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java:383]: Fixed

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java (hunk 2 above)
Comment (OneSizeFitsQuorum) [reply server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java:374]: Fixed

**Review by MrQuansy — APPROVED** [review]

**Pull request merged**